src/components/Badges/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react"
+
+import * as S from "./styles"
+import { Props } from "./types"
+
+import { Box } from "../Box"
+
+export const Badges = ({
+    label,
+    type = "badge",
+    startIcon,
+    endIcon
+}: Props) => {
+    const [showCloseIcon, setShowCloseIcon] = useState(false)
+    const [closeTag, setCloseTag] = useState(false)
+
+    return (
+        <Box>
+            {type === "tag" ? (
+                <S.Tags
+                    onMouseEnter={() => setShowCloseIcon(!showCloseIcon)}
+                    onMouseLeave={() => setShowCloseIcon(!showCloseIcon)}
+                    style={closeTag ? { display: "none" } : {}}
+                    data-testid='testid-tag'
+                >
+                    {label}
+                    {showCloseIcon ? (
+                        <S.CloseIcon onClick={() => setCloseTag(!closeTag)} data-testid='testid-icon' />
+                    ) : null}
```

Comment:
1. Adicionar atributo `id` para compor test-id do botao de fechar
2. Usar condicional && para simplificar lógica de exibir ícone
3. Adicionar prop `onDelete` e remove state local `closeTag`

```suggestion
                    {showCloseIcon && (
                        <S.CloseIcon onClick={onDelete} data-testid={`tag-${id}-delete-button`} />
                    )}
```